src/common/Gateway.sol

```diff
@@ -275,8 +273,15 @@ contract Gateway is Auth, Recoverable, IGateway {
             }
         }
 
-        batcher = address(0);
         _endBatching(refund);
+        _batcher = address(0);
```

Comment:
Shouldn't we check here `require(_batcher == address(0))`, to force `lockCallback` to have been called?

Reply:
Great idea!